cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/ArtifactSelector.java

```diff
@@ -55,9 +56,16 @@ public ArtifactSelector(String pluginType,
       msg.append(" name ");
       msg.append(name);
     }
-    if (version != null) {
-      msg.append(" version ");
-      msg.append(version.getVersion());
+    if (range != null) {
+      if (range.isExactVersion()) {
+        msg.append(" version ");
+        msg.append(range.getLower());
+      } else {
+        msg.append(" version lower range ");
```

Comment:
this is actually ambiguous because you can't tell if it's inclusive or exclusive. Instead of having an 'if' on whether it is an exact version, let's add a getVersionString() method that returns either the exact version or the '[lower,upper]' representation.